### ISSUE TITLE
Team sync / HTTP 500

### ISSUE THREAD

**Issue by jaymecd** [pullapprove/support#80]
Hi,

Team sync is not working for 2nd day,

> https://pullapprove.com/foodpanda/sync-teams/ => HTTP 500,
> Uh oh. There was a problem, we're looking into it. In the meantime, try refreshing.

Are you on it?

**Comment by davegaeddert** [comment]
Hey @jaymecd, I did actually see those errors coming through and had it on my list to look at today. It seems as though none of the foodpanda users who have signed into PullApprove are admins of foodpanda. I see you, jsafoodpanda and niko-fp. Maybe one of those users was an admin of foodpanda before but now is not?

**Comment by jaymecd** [comment]
@davegaeddert, hardly to say, I have admins access to couple repos only.
Is admin role is required to make this sync, even on my behalf?

**Comment by davegaeddert** [comment]
So syncing teams for the organization is a little tricky -- since some teams can be "secret", the only way to make sure we know about all of them is to sync on behalf of an owner/admin of the organization itself... Otherwise people might try to use teams in their .pullapprove.yml that PullApprove doesn't actually know about.

I made a change here so that if no org "admin" token is available, it will fall back to a org "member" token. The drawback is that it means you will **not be able to use "secret" teams** if it had to fall back to a member token. Reason being that we can't guarantee we know about all of the teams, and the ones we do know about would be dependent on which user made the call (since they can see all of their own teams, secret included).

Long story short, it should work for you now. I doubt you'll run into issues with secret teams but let me know if you do.

**Comment by jaymecd** [comment]
Confirmed, thank you very much.

**Comment by jaymecd** [comment]
@davegaeddert, same issue on repository level:

> https://pullapprove.com/foodpanda/shop-new/sync-teams/ => HTTP 500
